model/src/main/java/org/jboss/pnc/model/BuildEnvironment.java

```diff
@@ -19,6 +19,7 @@
 
 import java.util.HashMap;
 import java.util.Map;
+import java.util.Objects;
```

Comment:
I would also modify the unique constraint to be the same? systemImageId and systemImageRepositoryUrl

Reply:
I don't know.
The `systemImageId` is already unique so paired with anything (e.g. `systemImageRepositoryUrl`) it's still unique. At first I wanted to use only the `systemImageId` in the equals thinking that `systemImageRepositoryUrl` will be added later when the BuildEnvironment is refactored. But then I thought that it is not a problem to add it now and we won't forget to add it later. But I didn't plan to do any DB changes in this PR.
If you think new unique constraint should be added (and the one from `systemImageId` removed?) I'll do it.

Reply:
Yes it was more to catch 2 birds with 1 stone while you are making such changes. We might use the same systemimageID with different systemimagerepositoryurl soonish (for Quay). I am ok in making DB changes when they will be needed, it's just a bit strange that the code and DB have different uniqueness though.